util/NonparametricShift.py

```diff
@@ -4,6 +4,52 @@
 import torch
 import torch.nn as nn
 
+class Modified_NonparametricShift(object):
+    def _extract_patches_from_flag(self, img, patch_size, stride, flag, value, normed=True):
+        n_dim = 3
+        assert img.dim() == n_dim, 'image must be of dimension 3.'
+
+        kH, kW = patch_size, patch_size
+        dH, dW = stride, stride
+        input_windows = img.unfold(1, kH, dH).unfold(2, kW, dW)
+        
+        i_1, i_2, i_3, i_4, i_5 = input_windows.size(0), input_windows.size(1), input_windows.size(2), input_windows.size(3), input_windows.size(4)
+        
+        input_windows = input_windows.permute(1,2,0,3,4).contiguous().view(i_2*i_3, i_1, i_4, i_5)
+
+        ## EXTRACT MASK OR NOT DEPENDING ON VALUE
+        input_window = input_windows[flag == value]
+        input_window = input_window.view(input_window.size(0), -1)
+        
+        for i in range(input_window.size(0)):
+            input_window[i] = input_window[i]*(1/(input_window[i].norm(2)+1e-8))
+            return input_window
+
+    def _norm(self, input_window):
+        for i in range(input_window.size(0)):
+            input_window[i] = input_window[i]*(1/(input_window[i].norm(2)+1e-8))
+        return input_window        
+    
+    def _paste(self, img, patch_size, stride, flag, patch):
```

Comment:
`_paste` and `_extract_patches_from_flag` have duplicated code, `Line 34~Line 42` with `Line 9~18`. Can they share code?